src/components/Home/Home.js

```diff
@@ -1,29 +1,57 @@
-import React from 'react'
+import React, { useEffect, useState } from 'react'
 import BottomNav from '../Layouts/BottomNav'
 import Header from '../Layouts/Header'
 import Shop from './Shop'
+import axios from '../../axios'
+import { useStateValue } from '../../StateProvider'
 
 function Home () {
+  const [{ shops }, dispatch] = useStateValue()
+
+  const [latitude, setLatitude] = useState(null)
+  const [longitude, setLongitude] = useState(null)
+  const [error, setError] = useState(null)
+  const [loading, setLoading] = useState(true)
+
+  useEffect(() => {
+    navigator.geolocation.getCurrentPosition(function (position) {
+      setLatitude(position.coords.latitude)
+      setLongitude(position.coords.longitude)
+    })
+    if (longitude && latitude) {
+      shopsListGetRequest()
+    }
+  }, [latitude, longitude])
+
+  const shopsListGetRequest = async () => {
+    try {
+      const resp = await axios.get('/api/v1/shops-list', { params: { longitude: longitude, latitude: latitude } })
+      setLoading(false)
+      setError(null)
+      dispatch({
+        type: 'FETCH_SHOPS',
+        shopData: resp.data
+      })
+    } catch (err) {
+      setLoading(false)
+      setError(err.message)
+    }
+  }
```

Comment:
Take this function inside `useEffect` [like this](https://dmitripavlutin.com/react-useeffect-explanation/#5-fetching-data). Otherwise, it will keep on re-rendering.